### PR TITLE
Log watcher properly

### PR DESCRIPTION
**Pull request by michalpristas**
## What does this PR do?

This PR makes watcher log into `logs` directory.
Logs are name in a similar manner as normal logs but with `watcher` suffix (see picture)
image

When rollback is performed everything except logs directory is removed. this can then be collected later.
On successful upgrade full cleanup is performed and everything except latest agent is removed.

This PR also modifies diagnostics collection.
Now logs are stored into  `logs/elastic-agent-{hash}`
So in case of rollback watcher logs will be present in `oldhash` dir and agent logs will be store in `newhash` directory

## Why is it important?

Debug-ability of agent during rollback

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

Fixes: #2262

### CODE REVIEW

File: internal/pkg/diagnostics/diagnostics.go [hunk 1]
```diff
@@ -282,12 +312,14 @@ func zipLogs(zw *zip.Writer, ts time.Time) error {
...
-		return fmt.Errorf("failed to collect endpoint-security logs: %w", err)
+	if collectServices {
+		if err := collectServiceComponentsLogs(zw); err != nil {
+			return fmt.Errorf("failed to collect endpoint-security logs: %w", err)
+		}
 	}
 
 	// using Data() + "/logs", for some reason default paths/Logs() is the home dir...
-	logPath := filepath.Join(paths.Home(), "logs") + string(filepath.Separator)
+	logPath := filepath.Join(pathsHome, "logs") + string(filepath.Separator)
```
Comment (pchila) [inline internal/pkg/diagnostics/diagnostics.go:322]: Not introduced in this change but I am just curious: why do we need an extra separator ad the end of logPath ?

**Review by pchila — COMMENTED** [review]
Didn't test it yet but the change looks good
I don't see any automated testing for this, would it be possible to write some?

File: internal/pkg/diagnostics/diagnostics.go (hunk 1 above)
Comment (michalpristas) [reply internal/pkg/diagnostics/diagnostics.go:322]: dunno, probably something with a zip writer

**Review by blakerouse — APPROVED** [review]
Change looks good. Hopefully we can cover this code which is hard to unit test with an upgrade integration test.

**Comment by pierrehilbert** [comment]
> Change looks good. Hopefully we can cover this code which is hard to unit test with an upgrade integration test.

You know what to add to [your upgrade integration test then :-)](https://github.com/elastic/elastic-agent/issues/2176)

**Pull request merged**